lib/config/index.ts

```diff
@@ -113,6 +113,8 @@ export async function parseConfigs(
   });
   global.trustLevel = config.trustLevel || 'low';
   delete config.trustLevel;
+  global.allowedPostUpgradeTasks = config.allowedPostUpgradeTasks;
```

Comment:
I don't like global variables. We should move those to a shared config object or pass that with config as other options.

Reply:
Agreed. When I first read this I thought that 'self-hosted' settings were controlled by this global object. I didn't realise that they could be specified by the regular config object.